treap/treap.go

```diff
@@ -0,0 +1,145 @@
+// Package treap implements a treap (tree-heap) data structure
+// Each node hold a key and a random priority
+// Keys arranged as binary search tree (BST)
+// Priorities arranged as heap
+//
+// Structure is not thread safe.
+//
+// Reference: https://en.wikipedia.org/wiki/Treap
+
+package treap
+
+import (
+	"math/rand"
+	"time"
+
+	goheap "github.com/theodesp/go-heaps"
+)
+
+type Node struct {
+	Priority    goheap.Integer
```

Comment:
Could priority be an int here as we don't want to use the `goheap.Integer` helper

Reply:
Priority still use Compare method